extension/apikeyauthextension/authenticator_test.go

```
@@ -77,7 +77,7 @@ func TestAuthenticator(t *testing.T) {
 				},
 				Status: 400,
 			}),
-			expectedErr: `rpc error: code = Unauthenticated desc = error checking privileges for API Key "id": status: 400, failed: [a_type], reason: a_reason`,
+			expectedErr: `rpc error: code = Unavailable desc = error checking privileges for API Key "id": status: 400, failed: [a_type], reason: a_reason`,
```

It looks like the `newCannedErrorHandler` always returns a `types.ElasticsearchError`, do we need a handler that returns no type for this scenario: `error on the TCP connection level`?

---

How can an ES error be returned if ES is never reached?

---

Sorry my understanding from the below comment is that we want to test the scenario when no ES error type is returned. But this handler always returns an ES error type, so I was wondering if we need to return a different error type
```
// If no ES error type is found, it implies an error on the TCP connection level.
// In this case, we want to return an unavailable code so we have to option of
// handling this as a user-defined error later on.
```

---

My guess is that this test is passing due to the check being false: https://github.com/elastic/opentelemetry-collector-components/pull/797/files#diff-caf0d6c293198c76d9ba0005aa8a22d099b5baec8b5ea4d20d769c4fcbba3e35R285